RentallCarsAPI/Controllers/CarController.cs

```diff
@@ -18,86 +16,175 @@ public class CarController : Controller
+            {
+                response.Message = "File reading failed";
+                return BadRequest(response);
+            }
+            if (!ValidateParams(model))
+            {
+                response.Message = "Invalid Transmition or Mark";
+                return BadRequest(response);
+            }
+            var car = new Car();
+            car.Id = ++cars.Last().Id;
+            car.Transmition = model.Transmition;
+            car.Mark = model.Mark;
+            car.Model = model.Model;
+            car.Doors = model.Doors;
+            car.Color = model.Color;
+            cars.Add(car);
             try
             {
-                var cars = GetAll();
-                foreach (var aux in cars) 
-                {
-                    if (model.Id == aux.Id)
-                    {
-                        oResponse.Message = "Invalid Id, it´s used";
-                        return Ok(oResponse);
-                    }
-                }
-                if (((int)model.Mark >= 0 && (int)model.Mark <= 5) && ((int)model.Transmition >= 0 && (int)model.Transmition <= 1))
-                {
-                    var oCar = new Car();
-                    oCar.Id = model.Id;
-                    oCar.Transmition = model.Transmition;
-                    oCar.Mark = model.Mark;
-                    oCar.Model = model.Model;
-                    oCar.Doors = model.Doors;
-                    oCar.Color = model.Color;
-                    cars.Add(oCar);
-                    var writer = JsonConvert.SerializeObject(cars,Formatting.Indented);
-                    System.IO.File.WriteAllText("cars.txt", writer);
-                    oResponse.Succes = true;
-                    oResponse.Data = writer;
-                }
-                else 
-                {
-                    oResponse.Message = "Invalid Transmition or Mark";
-                }
+                var writer = JsonConvert.SerializeObject(cars, Formatting.Indented);
+                System.IO.File.WriteAllText("cars.txt", writer);
+                response.Succes = true;
+                response.Data = car;
+                response.Message = "Successfully added";
             }
-            catch(Exception ex)
+            catch (Exception ex)
             {
-                oResponse.Message = "Impossible to add";
+                response.Message = "Impossible to add";
+                return BadRequest(response);
             }
-
-            return Ok(oResponse);
-
+            return Ok(response);
         }
 
         [HttpGet("{id}")]
         public IActionResult Get(int id)
         {
-            var oResponse = new Response();
+            var response = new Response();
+             
+            var cars = GetAll();
+            if (cars == null)
+            {
+                response.Message = "File reading failed";
+                return BadRequest(response);
+            }
+            foreach (var aux in cars)
```

Comment:
Y si usamos un nombre mas representativo?
```suggestion
            foreach (var car in cars)
```

Reply:
perfecto, ya fue modificado.